src/test/java/de/tum/in/www1/artemis/FileIntegrationTest.java

```diff
@@ -199,17 +357,17 @@ void testGetLectureAttachment_mimeType() throws Exception {
         // add a new file type to allow to check the mime type detection in FileResource with an exotic extension
         fileResource.addAllowedFileExtension("exotic");
 
-        String filename = "attachment.exotic";
-        String attachmentPath = createLectureWithAttachment(filename, HttpStatus.CREATED);
+        Attachment attachment = createLectureWithAttachment("attachment.exotic", HttpStatus.CREATED);
+        String attachmentPath = attachment.getLink();
         // get access token and then request the file using the access token
-        String accessToken = request.get("/api/files/attachments/access-token/" + filename, HttpStatus.OK, String.class);
+        String accessToken = request.get(attachmentPath + "/access-token", HttpStatus.OK, String.class);
         String receivedAttachment = request.get(attachmentPath + "?access_token=" + accessToken, HttpStatus.OK, String.class);
         assertThat(receivedAttachment).isEqualTo("some data");
 
         fileResource.addRemoveFileExtension("exotic");
     }
 
-    private String createLectureWithAttachment(String filename, HttpStatus expectedStatus) throws Exception {
+    public Attachment createLectureWithAttachment(String filename, HttpStatus expectedStatus) throws Exception {
```

Comment:
Helper methods can probably be private, right?
```suggestion
    private Attachment createLectureWithAttachment(String filename, HttpStatus expectedStatus) throws Exception {
```

Reply:
Not sure why I changed it 😅